server/tserver/src/main/java/org/apache/accumulo/tserver/log/SortedLogRecovery.java

```diff
@@ -151,126 +98,155 @@ private String getPathSuffix(String pathString) {
+        checkState(entry.getKey().seq >= recoverySeq); // should only fail if bug elsewhere
+
+        if (entry.getKey().event == MUTATION) {
+          mr.receive(entry.getValue().mutations.get(0));
+        } else if (entry.getKey().event == MANY_MUTATIONS) {
+          for (Mutation m : entry.getValue().mutations) {
+            mr.receive(m);
+          }
+        } else {
+          throw new IllegalStateException("Non mutation event seen " + entry.getKey().event);
         }
       }
     }
-    if (tid < 0) {
-      throw new UnusedException();
-    }
+  }
 
-    log.debug("Found tid, seq " + tid + " " + defineKey.seq);
-
-    // Scan start/stop events for this tablet
-    key = defineKey;
-    key.event = COMPACTION_START;
-    reader.seek(key);
-    while (reader.next(key, value)) {
-      // LogFileEntry.printEntry(entry);
-      if (key.tid != tid)
-        break;
-      if (key.event == COMPACTION_START) {
-        if (lastStartToFinish.compactionStatus == Status.INITIAL)
-          lastStartToFinish.compactionStatus = Status.COMPLETE;
-        if (key.seq <= lastStartToFinish.lastStart)
-          throw new RuntimeException("Sequence numbers are not increasing for start/stop events: "
-              + key.seq + " vs " + lastStartToFinish.lastStart);
-        lastStartToFinish.update(fileno, key.seq);
-
-        // Tablet server finished the minor compaction, but didn't remove the entry from the
-        // METADATA table.
-        log.debug(
-            "minor compaction into " + key.filename + " finished, but was still in the METADATA");
-        if (suffixes.contains(getPathSuffix(key.filename)))
-          lastStartToFinish.update(-1);
-      } else if (key.event == COMPACTION_FINISH) {
-        if (key.seq <= lastStartToFinish.lastStart)
-          throw new RuntimeException("Sequence numbers are not increasing for start/stop events: "
-              + key.seq + " vs " + lastStartToFinish.lastStart);
-        if (lastStartToFinish.compactionStatus == Status.INITIAL)
-          lastStartToFinish.compactionStatus = Status.LOOKING_FOR_FINISH;
-        else if (lastStartToFinish.lastFinish > lastStartToFinish.lastStart)
-          throw new RuntimeException(
-              "COMPACTION_FINISH does not have preceding COMPACTION_START event.");
-        else
-          lastStartToFinish.compactionStatus = Status.COMPLETE;
-        lastStartToFinish.update(key.seq);
-      } else
-        break;
-    }
-    return tid;
+  Collection<String> asNames(List<Path> recoveryLogs) {
+    return Collections2.transform(recoveryLogs, new Function<Path,String>() {
+      @Override
+      public String apply(Path input) {
+        return input.getName();
+      }
+    });
   }
 
-  private void playbackMutations(MultiReader reader, int tid, LastStartToFinish lastStartToFinish,
+  public void recover(KeyExtent extent, List<Path> recoveryLogs, Set<String> tabletFiles,
```

Comment:
Personal preference to have public methods and entry points at the top of classes.  Unlike binary data, humans read from the top down :)